modules/mono/editor/Godot.NET.Sdk/Godot.NET.Sdk.sln

```diff
@@ -1,14 +1,17 @@
-
+﻿
 Microsoft Visual Studio Solution File, Format Version 12.00
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Godot.NET.Sdk", "Godot.NET.Sdk\Godot.NET.Sdk.csproj", "{31B00BFA-DEA1-42FA-A472-9E54A92A8A5F}"
+# Visual Studio Version 17
+VisualStudioVersion = 17.9.34728.123
+MinimumVisualStudioVersion = 10.0.40219.1
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Godot.NET.Sdk", "Godot.NET.Sdk\Godot.NET.Sdk.csproj", "{31B00BFA-DEA1-42FA-A472-9E54A92A8A5F}"
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Godot.SourceGenerators", "Godot.SourceGenerators\Godot.SourceGenerators.csproj", "{32D31B23-2A45-4099-B4F5-95B4C8FF7D9F}"
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Godot.SourceGenerators", "Godot.SourceGenerators\Godot.SourceGenerators.csproj", "{32D31B23-2A45-4099-B4F5-95B4C8FF7D9F}"
```

Comment:
This change is incorrect, same for all the other project GUIDs in this and other `.sln` files. See https://github.com/dotnet/project-system/issues/1821#issuecomment-388689744.

Reply:
I'm not convinced this is strictly incorrect, at least for C# specifically. The conversion from `FAE04EC0…` to `9A19103F…` seems to lineup with the GUIDs for `C# (Framework)` and `C# (.NET Core)` respectively[^1]. I have no clue *why* it does that conversion, as the old and new guids seem to work pretty much identically, but I'm inclined to keep it on this new guid since I haven't seen any reports of the conversion happening the other direction (that is: any future `.sln` changes won't have conversion headaches).

[^1]: https://github.com/JamesW75/visual-studio-project-type-guid?tab=readme-ov-file#by-guid

Reply:
Well, a Microsoft employee is saying in the issue I linked that the GUID that should be persisted for C# projects is `FAE04EC0-301F-11d3-BF4B-00C04F79EFBC`.

> I have no clue why it does that conversion

If you want more context, it seems Visual Studio changes the GUID for projects that explicitly opt-in into the new project system[^1]. But read the full thread[^2] if you want more details.

[^1]: https://github.com/dotnet/project-system/blob/main/docs/opening-with-new-project-system.md
[^2]: https://github.com/dotnet/project-system/issues/1821

Reply:
Seeing that Mac has issues with reverting automatically is enough to make me not want to use the new system after all. Still, this is a fairly old thread by this point; you'd think there'd be a more official post listing the proper guids somewhere.